### ISSUE TITLE
Expose a stream to listen for state changes from a reactive model

### ISSUE THREAD

**Issue by D10100111001** [GIfatahTH/states_rebuilder#78]
It would be great to have a stream that is added to whenever `setState` is called. This is necessary in situation where there is no access to context available (pure dart services). 

```dart
Injector.getAsReactive<MyConfig>().stream.listen((MyConfig updatedConfig) => updatedConfig.test);
```

**Comment by GIfatahTH** [comment]
The availability of the context is not a limiting factor in states_rebuilder.

If I understand you use case you can use OnSetStateListener:

```dart
OnSetStateListener(
          models: [Injector.getAsReactive<MyConfig>()],
          onData: (MyConfig updatedConfig) {
            updatedConfig.test;
          },
          onError: (error){

          },
          child: ...,
        )
```

**Comment by D10100111001** [comment]
Isn't `OnSetStateListener` a widget? I'm talking about having access to it in a class where there is no concept or coupling with widgets. It is a plain dart class service. It will not be rendered or ever be part of the widget tree or any real flutter flow.

**Comment by D10100111001** [comment]
@GIfatahTH Any suggestions?

**Comment by GIfatahTH** [comment]
At the time being,in the next release I will add and onData callback to the ReactiveModel, so you can 
```dart
RM.get<MyConfig>().onData((MyConfig updatedConfig) => updatedConfig.test);
```

The limitation is that the onData is defined once, that is if you define a second onData it will override the first.

**Comment by D10100111001** [comment]
@GIfatahTH Awesome, thank you.

**Comment by GIfatahTH** [comment]
solved in #90
